### ISSUE TITLE
[REGRESSION] Character classes are not working in globs in 1.36

### ISSUE THREAD

**Issue by ajssd** [microsoft/playwright#24316]
The 1.36.x release just says "Summer maintenance release".  Yet, simply upgrading from 1.35.1 to 1.36.0 has broken many of my tests (waitForURL is failing whereas it did not before).  It would be helpful if Playwright would document the specific changes that were made with each point release.

Specifically, what changed between 1.35.0 and 1.36.0?

**Comment by aslushnikov** [comment]
> The 1.36.x release just says "Summer maintenance release". Yet, simply upgrading from 1.35.1 to 1.36.0 has broken many of my tests (waitForURL is failing whereas it did not before).

@ajssd This sounds like an unintended regression. What exactly broke? Could you please come up with a repro that works in 1.35 and is broken in 1.36?

**Comment by ajssd** [comment]
The problem appears to be glob-handling in waitForURL.  A full reproducible case is difficult, but I have narrowed it down to this:

I have a frame with a URL like

http://mydomain:8080/blah/blah/three-columns/settings.html?id=settings-e3c58efe-02e9-44b0-97ac-dd138100cf7c&blah

The following, using regular expression works:
```
await contentFrame.waitForURL(new RegExp(`^.*/three-columns/settings\\.html\\?.*id=[a-z0-9].*$`));
```

The following, using simple globs, also work:

```
await contentFrame.waitForURL('**/three-columns/settings.html?**id=**');
await contentFrame.waitForURL('**/three-columns/settings.html?**id=?**');
```

But the following, using a simple range, does NOT work (this is the bug):
```
await contentFrame.waitForURL('**/three-columns/settings.html?**id=[a-z]**');
```

Aside: I would like to switch to using `expect(page).toHaveURL()` but that apparently does not support frames (only pages)... is that true?

**Comment by aslushnikov** [comment]
@ajssd Thank you for digging this. Turns out this is a regression caused by https://github.com/microsoft/playwright/pull/23932

>  I would like to switch to using expect(page).toHaveURL() but that apparently does not support frames (only pages)... is that true?

It doesn't support at the moment, but this sounds reasonable; could you please file this a separate feature request for this?

**Comment by aslushnikov** [comment]
This is fixed; the fix will be available in 1.36.2 during the next hour.
